crates/polars-parquet/src/parquet/compression.rs

```diff
@@ -95,13 +95,19 @@ pub fn compress(
         )),
         #[cfg(feature = "zstd")]
         CompressionOptions::Zstd(level) => {
```

Comment:
Can we add setting to the writer. E.g. that the streaming engine uses the bulk compresison, and the default writer uses the old style?

Reply:
It's not clear to me if that'd work, the Polars-level compression API (the `compress()`) is not designed for streaming, it has no state...

Reply:
I don't follow. I understand that his new snippet is faster in the streaming engine, but slower in the default (single batch) case.

So I assume we can set a flag on the reader and use that to branch here.

Reply:
Ah, I misunderstood what you meant.  I could do that, or maybe... size-based heuristic. In which case I would initially log the size of data, which might also be informative for the batch-size hypothesis.

Reply:
`sink_parquet()` ends up compressing chunks of sizes 1K to 27K.

`write_parquet()` mostly writes chunks of sizes 130K to 680K.

So there's a clear divide here which plausibly explains the performance differences: an accumulation of doing lots of small operations will add overhead in a variety of both easy (this PR) and difficult to observe ways (e.g. worse branch prediction).

For the remaining #11699 slowness, investigating batch sizes further seems like a good idea. For this PR I'll see if a size-based heuristic speeds things up.

Reply:
After doing more benchmarking runs, I think the difference in `COLLECT+WRITE` is just noise: runtime varies quite a lot in both `main` and this branch, with no discernible pattern about which is slower or faster. So I think this PR should be OK as is.

Reply:
Alright. Thanks. It is a huge improvement.